src/styles/foundation/_layout.scss

```diff
@@ -60,7 +56,7 @@ $layout-width-data: (
 $dismiss-icon-size: 32px;
 
 @function top-bar-height() {
-  @return rem(56px);
+  @return var(--p-top-bar-height, rem(56px));
```

Comment:
Currently it's possible to do stuff like `some-property: top-bar-height() - 10px;` and it'll work because it returns a plain number. 

This change will break that behaviour, which feels pretty scary. It doesn't look like we use it in that way internally but external folk might.

If this is super useful and simplifies a bunch of stuff (looks like it'd help further simplify some of the bits I touched in https://github.com/Shopify/polaris-react/pull/3158/files) then it should be a new function. That new function should live in a new file, and be imported in _common, but not _public-api so we don't end up exposing it publicly (and thus would be free to make this sort of change in the future because we don't need to care about external consumers).

Reply:
Excellent point. I don't think this is necessary for this PR so I can revert